src/rgw/rgw_restore.cc

```diff
@@ -655,6 +701,18 @@ int Restore::restore_obj_from_cloud(rgw::sal::Bucket* pbucket,
   }
 
   ldpp_dout(this, 10) << __PRETTY_FUNCTION__ << ": Restore of object " << pobj->get_key() << " is in progress." << dendl;  
+
```

Comment:
is "restore_obj_from_cloud()" called from a REST API?
if so, you need to split the "reserve/commit" of the notification, so that the notification reliably sent, and the client should get an error if the notification queue is full.

when sending notifications from a background process, the return value may be ignored, unless we want to include it in the retry mechanism of the background process.

Reply:
see comment from the tracker:
```
note that in case of "post" there is an actual reserve/commit process (similar to other notification sending in rgw_op.cc) since there is an actual client to send reply to.
while the other cases we ignore any "reserve" failures, similar to other lifecycle notifications.
```